File: 프로그래머스/1/161990. 바탕화면 정리/바탕화면 정리.java
import java.util.*;

class Solution {
    public int[] solution(String[] wallpaper) {
        int[] answer = {};
        
        ArrayList<Integer> rList = new ArrayList<>();
        ArrayList<Integer> cList = new ArrayList<>();
        
        for(int i = 0; i < wallpaper.length; i++){
            for(int j = 0; j < wallpaper[i].length(); j++){
                if(wallpaper[i].charAt(j) == '#'){
                    rList.add(i);
                    cList.add(j);
                }
            }
        }
        
        Collections.sort(rList);
        Collections.sort(cList);
        
        int minR = rList.get(0); // 가장 작은 값
        int minC = cList.get(0); // 가장 작은 값
        int maxR = rList.get(rList.size() - 1); // 가장 큰 값
        int maxC = cList.get(cList.size() - 1); // 가장 큰 값
        
        return new int[]{minR, minC, maxR + 1, maxC + 1};
    }
}

File: 프로그래머스/1/161990. 바탕화면 정리/바탕화면 정리.py
def solution(wallpaper):
    
    files = []
    
    for i, w in enumerate(wallpaper):
        for j in range(len(w)):
            if w[j] == "#":
                files.append((i,j))
                
    min_r, min_c = float("inf"), float("inf")
    max_r, max_c = float("-inf"), float("-inf")

    for file in files:
        r = file[0]
        c = file[1]
        
        if r < min_r: min_r = r
        if r > max_r: max_r = r
        if c < min_c: min_c = c
        if c > max_c: max_c = c
    
    return [min_r, min_c, max_r + 1, max_c + 1]